src/risk_atlas_nexus/library.py

```diff
@@ -98,182 +101,242 @@ def get_all_risks(cls, taxonomy=None):
             list[Risk]
                 Result containing a list of AI risks
         """
-        if taxonomy is not None and type(taxonomy) != str:
-            raise ValueError("Taxonomy must be a string", taxonomy)
+        type_check("<RAN_TYPE_CHECK_ERROR>", str, allow_none=True, taxonomy=taxonomy)
 
         risk_instances = cls._risk_explorer.get_all_risks(taxonomy)
         return risk_instances
 
-    def get_risk_by_tag(cls, tag):
-        """Get risk definition from the LinkML, filtered by risk atlas tag
+    def get_risk(cls, tag=None, id=None, name=None, taxonomy=None):
+        """Get risk definition from the LinkML, filtered by risk atlas id, tag, name
 
         Args:
-            tag: str
-                The string label identifying the risk
+            id: (Optional) str
+                The string ID identifying the risk
+            tag: (Optional) str
+                The string tag identifying the risk
+            name: (Optional) str
+                The string name identifying the risk
             taxonomy: str
                 (Optional) The string label for a taxonomy
 
         Returns:
             Risk
                 Result containing a list of AI risks
         """
-        risk = cls._risk_explorer.get_risk_by_tag(tag=tag)
-        return risk
-
-    def get_risk_by_id(cls, id):
-        """Get a risk definition from the LinkML, filtered by risk id
-
-        Args:
-            id: str
-                The string label identifying the risk
+        type_check(
+            "<RAN_TYPE_CHECK_ERROR>",
+            str,
+            allow_none=True,
+            tag=tag,
+            id=id,
+            name=name,
+            taxonomy=taxonomy,
+        )
+        value_check(
+            "<RAN_VALUE_CHECK_ERROR>",
```

Comment:
Same as above. Are these codes just for the display or they get convert into a message?